src/ui/api/repository.go

```
@@ -154,39 +154,53 @@ func getRepositories(query *models.RepositoryQuery) ([]*repoResp, error) {
 		return nil, err
 	}
 
-	return assembleRepos(repositories)
+	return assembleReposInParallel(repositories), nil
 }
 
-func assembleRepos(repositories []*models.RepoRecord) ([]*repoResp, error) {
-	result := []*repoResp{}
+func assembleReposInParallel(repositories []*models.RepoRecord) []*repoResp {
```

I don't think we should eat errors here.